### PR TITLE
Fix 90*60 Console functions not working

### PR DESCRIPTION
**Pull request by BlitzWolfMatthew**
The title explains explains everything.
Need a review so I know what to change, the code is a bit messy right now.

### CODE REVIEW

File: Tests/Cosmos.TestRunner.Full/TestKernelSets.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public static IEnumerable<Type> GetKernelTypesToRun()
         // Stable kernel types: the ones that are stable and will run in AppVeyor
         public static IEnumerable<Type> GetStableKernelTypes()
         {
+            yield return typeof(ConsoleTest.Kernel);
```
Comment (quajak) [inline Tests/Cosmos.TestRunner.Full/TestKernelSets.cs:18]: Console test should not be enabled for appveyor testrunner.

File: Tests/Kernels/ConsoleTest/Kernel.cs [hunk 2]
```diff
@@ -32,100 +32,119 @@ protected override void Run()
         {
             try
             {
-                Console.Clear();
```
Comment (quajak) [inline Tests/Kernels/ConsoleTest/Kernel.cs:35]: Please rework the Console tests. I think we should run a series of tests for each resolution, similar to how its done in Graphics Test.

**Comment by BlitzWolfMatthew** [comment]
#1984

File: source/Cosmos.System2_Plugs/System/ConsoleImpl.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public static class ConsoleImpl
         private static Encoding ConsoleInputEncoding = Encoding.ASCII;
         private static Encoding ConsoleOutputEncoding = Encoding.ASCII;
 
-        private static readonly Cosmos.System.Console mFallbackConsole = new Cosmos.System.Console(null);
+        public static Cosmos.System.Console mFallbackConsole = new Cosmos.System.Console(null);
```
Comment (quajak) [inline source/Cosmos.System2_Plugs/System/ConsoleImpl.cs:18]: Why is this public?

File: source/Cosmos.System2_Plugs/System/ConsoleImpl.cs [hunk 4]
```diff
@@ -570,7 +570,55 @@ public static void SetWindowPosition(int left, int top)
...
+            }
+            else
+            {
+                throw new Exception("Invalid text size.");
+            }
+            mFallbackConsole.Cols = mFallbackConsole.mText.Cols;
+            mFallbackConsole.Rows = mFallbackConsole.mText.Rows;
+
+            //VGA memory
+            ((Cosmos.HAL.TextScreen)mFallbackConsole.mText).IO.Memory = new Cosmos.Core.MemoryBlock(0xB8000, (uint)(width * height * 2));
```
Comment (quajak) [inline source/Cosmos.System2_Plugs/System/ConsoleImpl.cs:617]: I think its better if we add a method SetWindowSize to TextScreen and let TextScreen do the necessary changes internaly

**Review by quajak — CHANGES_REQUESTED** [review]

File: Tests/Kernels/ConsoleTest/Kernel.cs [hunk 5]
```diff
@@ -32,100 +32,136 @@ protected override void Run()
...
             catch (Exception e)
             {
                 mDebugger.Send("Exception occurred: " + e.Message);
                 mDebugger.Send(e.Message);
                 TestController.Failed();
-            }
+            }            
+        }
+
+        public void TestConsoleEncoding()
```
Comment (zarlo) [inline Tests/Kernels/ConsoleTest/Kernel.cs:49]: the ```Console.ReadKey();``` in this method on line 84, 102 why are they there?

File: source/Cosmos.HAL2/TextScreen.cs [hunk 6]
```diff
@@ -46,8 +46,16 @@ public TextScreen()
             TextScreenHelpers.Debug("End of TextScreen..ctor");
         }
 
-        public override ushort Rows { get { return 25; } }
-        public override ushort Cols { get { return 80; } }
+        public void UpdateWindowSize()
```
Comment (zarlo) [inline source/Cosmos.HAL2/TextScreen.cs:49]: check the values of Rows and Cols as anyone can set the values

**Review by zarlo — CHANGES_REQUESTED** [review]

File: Tests/Kernels/ConsoleTest/Kernel.cs (hunk 5 above)
Comment (BlitzWolfMatthew) [reply Tests/Kernels/ConsoleTest/Kernel.cs:49]: This is how I found it, I didnt write that test. I only chnaged "germanic" to "german".

Comment (quajak) [reply Tests/Kernels/ConsoleTest/Kernel.cs:49]: I dont think these tests are meant to be run automatic and the ReadKeys allow the developer to check if the output looks correct.

File: source/Cosmos.HAL2/TextScreen.cs [hunk 7]
```diff
@@ -46,8 +46,16 @@ public TextScreen()
...
+        {
+            IO.Memory = new Cosmos.Core.MemoryBlock(0xB8000, (uint)(Cols * Rows * 2));
+            mRAM = new Cosmos.Core.MemoryBlock(0xB8000, (uint)(Cols * Rows * 2)).Bytes;
```
Comment (quajak) [inline source/Cosmos.HAL2/TextScreen.cs:52]: Nitpick: `mRAM  = IO.Memory.Bytes;`

**Review by quajak — CHANGES_REQUESTED** [review]
Please add the checks @zarlo suggested and fix the nitpick. The rest looks good.

File: source/Cosmos.HAL2/TextScreen.cs (hunk 6 above)
Comment (BlitzWolfMatthew) [reply source/Cosmos.HAL2/TextScreen.cs:49]: Rows and Cols are public but mFallBackConsole is private thus it cannot be accessed and changed by the user

**Review by quajak — APPROVED** [review]

**Pull request merged**